azkaban-common/src/main/java/azkaban/project/JdbcDependencyManager.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2019 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package azkaban.project;
+
+import azkaban.db.DatabaseOperator;
+import azkaban.spi.Dependency;
+import azkaban.spi.Storage;
+import azkaban.spi.FileValidationStatus;
+import azkaban.utils.HashUtils;
+import azkaban.utils.InvalidHashException;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Provides methods for interacting with dependency validation cache in DB. Used during thin archive
+ * uploads.
+ */
+@Singleton
+public class JdbcDependencyManager {
+  private static final Logger log = LoggerFactory.getLogger(JdbcDependencyManager.class);
+
+  private final DatabaseOperator dbOperator;
+
+  @Inject
+  JdbcDependencyManager(final DatabaseOperator dbOperator) {
+    this.dbOperator = dbOperator;
+  }
+
+  public Map<Dependency, FileValidationStatus> getValidationStatuses(final Set<Dependency> deps,
+      final String validationKey) throws SQLException {
+    Map<Dependency, FileValidationStatus> depValidationStatuses = new HashMap<>();
+    if (deps.size() == 0) {
+      // There's nothing for us to do.
+      return depValidationStatuses;
+    }
+
+    // Map of (filename + sha1) -> Dependency for resolving the dependencies already cached in the DB
+    // after the query completes.
+    Map<String, Dependency> hashAndFileNameToDep = new HashMap<>();
+
+    PreparedStatement stmnt = this.dbOperator.getDataSource().getConnection().prepareStatement(
+        "select file_name, file_sha1, validation_status from validated_dependencies where validation_key = ? and ("
+            + makeStrWithQuestionMarks(deps.size()) + ")");
```

Comment:
Can you parameterize this query or use String.format() instead? it's a little hard to read.